VHAR-6717 - Sanktiotyyppien hallinnan refaktorointia ja parantelua

K1ll3rF0x opened the pull request.

K1ll3rF0x committed:
Refaktoroi sanktiotyyppi-taulun logiikkaa
* Siirretty laji->tyyppi hierarkian määritys domain koodiin (cljc)
* Lisätty uniikit koodit sanktiotyypeille
* Pudotettu pois urakkatyyppi ja sanktiolaji sarakkeet
sanktiotyyppitaulusta
* Nimetty uudelleen osa sanktiotyypeistä
* Merkitty poistetuksi osa sanktiotyypeistä

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'develop' into VHAR-6717

K1ll3rF0x committed:
Korjaa kyselyiden TODO-kommentit

RelNotes:
Related:

K1ll3rF0x committed:
Hae urakalle kuuluvat sanktiolajit mapista, lisää raakaversio
sanktiotyyppien hausta sanktioiden lomakkeelle

RelNotes:
Related:

K1ll3rF0x committed:
Näytä sanktiolajin sanktiotyypit sanktiolomakkeella

RelNotes:
Related:

K1ll3rF0x committed:
Mahdollista sanktioiden tallentaminen ja näyttäminen taas
sanktiolomakkeella

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa laatupoikkeaman tallennus ja näyttäminen
* Lisätty uudentyyppinen sanktiolajien ja tyyppien hallinta

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa ylläpidon urakoiden sanktioiden tallennus lomakkeella, korjaile
sanktioihin liittyviä yksikkötestejä

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa typo

RelNotes:
Related:

K1ll3rF0x committed:
Siirrä laatupoikkeaman sanktiolajit yleiseen apufunktioon ja käytä samaa
paikkaa app tilassa, josta sanktiolajit haetaan.
* Laatupoikkeaman ja sanktiolomakkeen sanktiolajit haetaan nyt samasta
paikasta appin tilasta, eikä laatupoikkeaman näkymässä ole poikkeavaa
toimintaa. Sanktioiden domain-koodissa hallitaan myös laatupoikkeamien
sanktiolajeja urakan tyypin perusteella.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa sanktioraportti-testi ja sanktiot.sql testidata
* Haetaan sanktiot.sql:ssä sanktiotyypit uniikin koodin, eikä nimen
perusteella

RelNotes:
Related:

jpluomajoki-solita commented on src/cljc/harja/domain/laadunseuranta/sanktio.cljc:
```diff
@@ -1,4 +1,97 @@
+
+    ;; TODO: Varmistettava onko muita ehtoja
+
+    :else [:A :B :C]))
```
Muistutus ja arvonvähennyssanktio tulee myös laatupoikkeamiin

K1ll3rF0x committed:
Korjaa sanktioraportteihin liittyvät testit ja testidata, tee
laadunseurantatesteistä robustimpia, poista suorat ID-viittaukset
sanktiotyyppiin laadunseurantatesteistä.

RelNotes:
Related:

K1ll3rF0x committed:
Lisää muistutukset ja arvonvähennyssanktio laatupoikkeamien
sanktiolajivalintaan

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa laadunseuranta domain yksikkötestit

RelNotes:
Related:

K1ll3rF0x committed:
Lisää sanktiotyyppien urakkakohtaiset poikkeukset
* Tällä hetkellä riippumatta urakan tyypistä, ainoastaan urakan
alkupäivämäärän perusteella.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa laatupoikkeaman sanktiotyypin valinta, poista sorttaus
valintakomponentista
* Sorttausta ei tarvita käyttöliittymän puolella, koska tarkoitus on
määritellä sanktiotyppien järjesteys suoraan oikeaksi
sanktiolaji->sanktiotyyppi-koodi mapissa sanktiot-domainissa.

RelNotes:
Related:

jpluomajoki-solita commented on src/cljs/harja/tiedot/urakka.cljs:
```diff
@@ -486,15 +488,22 @@
-                                    (= :vesivayla-sanktiot vv-ls-sivu)))
-                (k/post! :hae-urakkatyypin-sanktiolajit {:urakka-id (:id urakka)
-                                                         :urakkatyyppi (:tyyppi urakka)}))))
+
```
Tästä voisi poistaa yhden tai kaksi tyhjää riviä

jpluomajoki-solita commented on src/cljs/harja/views/urakka/laadunseuranta/sanktiot.cljs:
```diff
@@ -31,7 +30,8 @@
             [harja.ui.valinnat :as valinnat]
-            [harja.ui.liitteet :as liitteet]))
+            [harja.ui.liitteet :as liitteet]
+            [harja.ui.debug :as debug]))
```
käyttämätön require?

jpluomajoki-solita reviewed (commented):
Hyvältä näyttää. Muutama koodinsiistimishuomio, mutta ei sen kummempaa. Voidaan puolestani mergettää kunhan todo-kommentit on tarkistettu.

K1ll3rF0x committed:
Korjaa requireja sanktiot-näkymässä, poista vanhoja duplikaatteja
requireja

RelNotes:
Related:

K1ll3rF0x committed:
Korjaile rivinvaihtoja

RelNotes:
Related:

K1ll3rF0x committed:
Poista vanhoja duplikaatteja requireja

RelNotes:
Related:

K1ll3rF0x committed:
Nimeä migraatio uudelleen

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into VHAR-6717

K1ll3rF0x committed:
Poista karttakomponentti Sanktiot-näkymästä

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa sanktiolomakkeen sanktiotyypin valinta vanhoille
sanktiokirjauksille, jotka ovat käyttäneet poistettuja 'Talvihoito' tai
'Muu tuote' sanktiotyyppejä.
* Näytetään sanktiotyypin valinta pelkkänä tekstinä lomakkeen
lukutilassa. Aiemmin käytössä oli valintakomponentti, joka oli asetettu
lukutilaan. Tämän takia lukutilaan ei tullut mitään sanktiotyyppiä
näkyville, koska poistetut sanktiotyypit on poistettu myös yleisesti
käyttöliittymän valinnoista.
* Muutettu myös kulunkohdistus myös pelkäksi tekstikomponentiksi
lukutilassa.
* Lisätty testidataan Oulun MHU:lle 'vanha sanktiokirjaus', joka käyttää
poistettua Talvihoito-sanktiotyyppiä.

RelNotes:
Related:

K1ll3rF0x committed:
Korjaa tilannekuva test, lisätty uusi laatupoikkeama

RelNotes:
Related:

K1ll3rF0x committed:
Päivitä maksurat_test summa, koska lisätty uusi sanktio testiaineistoon

RelNotes:
Related:

K1ll3rF0x committed:
Merge branch 'develop' into VHAR-6717
